sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -35,6 +40,9 @@
   /** Class of the original {@link DoFn} from which this signature was produced. */
   public abstract Class<? extends DoFn<?, ?>> fnClass();
 
+  /** Whether this {@link DoFn} does a bounded amount of work per element. */
+  public abstract PCollection.IsBounded isBounded();
```

Comment:
Good point. Renamed.
Also, I considered putting it on ProcessElement method, but I think it makes more sense on the DoFn because in theory (even if not in the current API) you could imagine a DoFn that splits an element into an unbounded number of restrictions, but does bounded work in a ProcessElement call, so the total amount of work is unbounded.